### DISCUSSION TITLE
Preserve msg

### DISCUSSION THREAD

**Discussion by mlconnor** [allanbunch/node-red-openai-api#19] [General]
Loving the library so far, thanks so much for building it. My question is this. I have a bunch of users that will be using this thing at the same time via websockets. I store a session Id in the message that I pass around so I know which socket to respond to.

**Question**: The node-red-openai-api library, while streaming, creates a new message for each chunk that comes back. I can't think of a way to recover the session Id or the other breadcrumbs I stored in the msg along the way. I was hoping that the library would preserve non-payload properties of the msg. I can update the library but I thought surely someone else must be doing multi-user with a way to solve this issue.

Thanks in advance! -Michael

**Comment by allanbunch** [comment]
Great to hear that you're finding value in the node. Your use case sounds interesting, and I understand your need for preserving non-payload message properties in chat response streams. Stay tuned for an update on this feature!

**Reply by mlconnor** [reply]
thanks Alan. it would nice to clone the original msg and then swap the payload with the chuck as you do today. happy to try to find the code and propose a change if it would help.

**Reply by allanbunch** [reply]
Hey Michael, I just released an update (v. 0.7.0) that covers your request. Please do let me know if you discover otherwise. Thanks for the nudge. This is definitely and improvement.

**Reply by allanbunch** [reply]
@mlconnor - I introduced a bug in v. 0.7.0 that causes Node-RED to get tripped up on distinguishing incoming chat completion message stream chunks. I've identified the problematic code and will release an update today.

**Reply by mlconnor** [reply]
Allan, thanks so much for diving into this. I'm excited to try the new version. I'll give it a shot this week. Thanks for your work on such a great library. It is easy to work around bugs like this using a custom function so I just use your library for most stuff and then write a function when needed. I don't like custom functions in general though so it will be nice to strip those out with your new release. Have a good weekend!
